mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/handlers/JobDiscoveryRouteHandlerAkkaImpl.java

```diff
@@ -77,13 +81,20 @@ public JobDiscoveryRouteHandlerAkkaImpl(ActorRef jobClustersManagerActor, Durati
             .maximumSize(500)
             .buildAsync(this::lastSubmittedJobId);
 
+        lastLaunchedJobIdStreamRespCache = Caffeine.newBuilder()
+            .expireAfterWrite(5, TimeUnit.SECONDS)
```

Comment:
umm, it should have been the same as before when job discovery (rely on /jobDiscoveryInfo) would switch before downstream consumers (rely on this endpoint with cache).
I don't suppose the publishers/subscribers did an additional check to ensure the job is in launched state before starting to send/consume data.

Reply:
just thinking out loud here: is it possible to subscribe this cache to get the update when there is a change while having a longer validation duration?

Reply:
on second look, it shouldn't matter because the propagation of jobId happens via BehaviorSubject and it should be instantaneous, the cache is merely refreshing response that contains the subject.